accord-core/src/test/java/accord/utils/Gens.java

```diff
@@ -90,6 +91,63 @@ public static <T> Gen<T> oneOf(Map<Gen<T>, Integer> values)
         return rs -> gen.next(rs).next(rs);
     }
 
+    public static <T> OneOfBuilder<T> oneOf()
+    {
+        return new OneOfBuilder<>();
+    }
+
+    public static class OneOfBuilder<T>
+    {
+        private final Map<Gen<T>, Integer> weighted = new LinkedHashMap<>();
+        private final Set<Gen<T>> unweighted = new LinkedHashSet<>();
+        private Gen.IntGen unknownWeightGen = Gens.ints().between(1, 10);
+
+        public OneOfBuilder<T> add(Gen<T> gen)
+        {
+            unweighted.add(gen);
+            return this;
+        }
+
+        public OneOfBuilder<T> add(int weight, Gen<T> gen)
+        {
+            weighted.put(gen, weight);
+            return this;
+        }
+
+        public OneOfBuilder<T> unknownWeights(Gen.IntGen gen)
+        {
+            this.unknownWeightGen = gen;
+            return this;
+        }
+
+        public Gen<Gen<T>> buildWithDynamicWeights()
```

Comment:
this is currently dead code... I didn't want to expand the usage of this class but figured the `CommandsBuilder` would be a good place as it does this pattern